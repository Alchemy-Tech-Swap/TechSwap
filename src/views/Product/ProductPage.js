import React, { useState, useEffect } from 'react';
import ProductsCard from '../../components/Products/ProductsCard';
import { fetchProducts } from '../../services/products';

export default function ProductPage() {
  const [products, setProducts] = useState([]);

  useEffect(() => {
    const getProducts = async () => {
      const data = await fetchProducts();
      setProducts(data);
    };

    getProducts();
  }, []);

  return (
    <div>
      <ProductsCard products={products} />
    </div>
  );
}
