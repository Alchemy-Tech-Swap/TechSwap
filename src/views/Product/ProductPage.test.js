import ProductPage from './ProductPage';
import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';

test('Products should render', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/']}>
      <Route path="/">
        <ProductPage
          products={{
            id: 34,
            created_at: '2022-01-21T01:20:34.936627+00:00',
            title: 'LogiTech MX Master 3',
            description: 'High performance computer mouse',
            price: 45,
            image:
              'https://epowwlsvajvjgzuvqqfe.supabase.co/storage/v1/object/public/product-image/public/logitech-mx-master-3-hero-alt.jpg',
            category: 'accessories',
            user_id: 23,
            sold: null,
            condition: 'Slightly used',
          }}
        />
      </Route>
    </MemoryRouter>
  );

  await screen.findByText('LogiTech MX Master 3');

  expect(container).toMatchSnapshot();
});
