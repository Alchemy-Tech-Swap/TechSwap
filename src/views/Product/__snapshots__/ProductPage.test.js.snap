// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products should render 1`] = `
<div>
  <div>
    <div>
      <h1
        class="productListTitle"
      >
        <img
          alt="Techswap2"
          class="techlogo"
          src="tswapbanner.png"
        />
        Welcome To The TechSwap
      </h1>
      <div
        class="productLists"
      >
        <a
          class="productLinks"
          href="/products/26"
        >
          <img
            class="productImage"
            src="https://epowwlsvajvjgzuvqqfe.supabase.co/storage/v1/object/public/product-image/public/desk.jpg"
          />
          <h1
            class="productTitle"
          >
            Standing Desk
          </h1>
          <p
            class="productCondition"
          >
            Used
          </p>
          <p
            class="productPrice"
          >
            $
            100
          </p>
        </a>
      </div>
      <div
        class="productLists"
      >
        <a
          class="productLinks"
          href="/products/34"
        >
          <img
            class="productImage"
            src="https://epowwlsvajvjgzuvqqfe.supabase.co/storage/v1/object/public/product-image/public/logitech-mx-master-3-hero-alt.jpg"
          />
          <h1
            class="productTitle"
          >
            LogiTech MX Master 3
          </h1>
          <p
            class="productCondition"
          >
            Slightly used
          </p>
          <p
            class="productPrice"
          >
            $
            45
          </p>
        </a>
      </div>
      <div
        class="productLists"
      >
        <a
          class="productLinks"
          href="/products/24"
        >
          <img
            class="productImage"
            src="https://epowwlsvajvjgzuvqqfe.supabase.co/storage/v1/object/public/product-image/public/intel.jpg"
          />
          <h1
            class="productTitle"
          >
            Intel Computer
          </h1>
          <p
            class="productCondition"
          >
            Used
          </p>
          <p
            class="productPrice"
          >
            $
            2345
          </p>
        </a>
      </div>
      <div
        class="productLists"
      >
        <a
          class="productLinks"
          href="/products/28"
        >
          <img
            class="productImage"
            src="https://www.lg.com/us/images/monitors/md06003957/gallery/1100-3.jpg"
          />
          <h1
            class="productTitle"
          >
            Marks Monitor
          </h1>
          <p
            class="productCondition"
          >
            NANI
          </p>
          <p
            class="productPrice"
          >
            $
            10000
          </p>
        </a>
      </div>
      <div
        class="productLists"
      >
        <a
          class="productLinks"
          href="/products/30"
        >
          <img
            class="productImage"
            src="https://epowwlsvajvjgzuvqqfe.supabase.co/storage/v1/object/public/product-image/public/152137-laptops-review-apple-macbook-pro-2020-review-image1-pbzm4ejvvs.jpg"
          />
          <h1
            class="productTitle"
          >
            Macbook Pro
          </h1>
          <p
            class="productCondition"
          >
            Used
          </p>
          <p
            class="productPrice"
          >
            $
            123
          </p>
        </a>
      </div>
    </div>
  </div>
</div>
`;
